src/Mappers/ModelMapper.php

```diff
@@ -48,15 +49,20 @@ public function export(array $records): array
      * @param bool  $persist           Whether to persist the parsed records
      *
      * @return array
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      */
     public function import(array $definitions, array $records, array $defaultAttributes = [], $persist = true): array
     {
         $imported = [];
-        $recordsByHandle = ArrayHelper::index($records, 'handle');
+        $recordsByHandle = [];
         foreach ($definitions as $handle => $definition) {
             $modelClass = $definition['class'];
             $converter = Craft::$app->controller->module->getConverter($modelClass);
             if ($converter) {
+                if (!count($recordsByHandle)) {
+                    $recordsByHandle = ArrayHelper::index($records, $converter->handle);
```

Comment:
This would use the first 'handle' field found for all records.
The converter can differ per record.

Reply:
True, but 'handle' was hardcoded before. This probably needs a bit of refactoring for it to be neat. Will look into it.